libpromises/processes_select.c

```diff
@@ -601,7 +624,14 @@ static bool SplitProcLine(const char *line,
         return false;
     }
 
-    Log(LOG_LEVEL_DEBUG, "Trying to match ps line '%s'", line);
+    size_t linelen = strlen(line);
+
+    if (LogGetGlobalLevel() >= LOG_LEVEL_DEBUG)
+    {
+        Log(LOG_LEVEL_DEBUG, "Parsing ps line: '%s'", line);
```

Comment:
I added it specifically because I was having trouble seeing where a line ended, since whitespace is supposed to be eaten by the parser (this is one of the tests), so I'd like to keep it.